### PR TITLE
Fix double attachment issue

### LINKED ISSUE #194: Sending attachment results in two messages per attachment.

### PR DESCRIPTION
**Pull request by shubhsherl**
Closes #194 
#210 includes this PR, please merge that one, after reviewing it.

### CODE REVIEW

**Comment by ear-dev** [comment]
@Shailesh351 Do you mind reviewing this please?  I tested it, and it works, but there is a lot of stuff here that I'm not yet familiar with and I think your review might be helpful.  Thank you!

File: app/lib/server/methods/deleteMessage.js [hunk 1]
```diff
@@ -28,7 +28,11 @@ Meteor.methods({
 			},
 		});
 
-		if (!originalMessage || !canDeleteMessage(uid, originalMessage)) {
+		if (!originalMessage) {
+			return;
+		}
+
+		if (!canDeleteMessage(uid, originalMessage)) {
```
Comment (Shailesh351) [inline app/lib/server/methods/deleteMessage.js:35]: @shubhsherl  Is this change needed? In upstream it is not there

File: app/lib/server/methods/sendMessage.js [hunk 2]
```diff
@@ -22,20 +21,7 @@ export function executeSendMessage(uid, message) {
 		});
 	}
 
-	if (message.ts) {
```
Comment (Shailesh351) [inline app/lib/server/methods/sendMessage.js:25]: Same here. This code is there in upstream

File: app/message-attachments/client/messageAttachment.js [hunk 3]
```diff
@@ -41,15 +41,8 @@ Template.messageAttachment.helpers({
 		return true;
 	},
 	uploading() {
-		return Template.parentData().msg.uploads;
+		return Template.parentData(2).msg.uploads;
```
Comment (Shailesh351) [inline app/message-attachments/client/messageAttachment.js:44]: Why '2' here? FYI in the upstream whole uploading function is removed

**Review by Shailesh351 — COMMENTED** [review]
@shubhsherl Added some comment. Please let me know if I'm missing anything.
Rest of the code looks good to me

File: app/lib/server/methods/deleteMessage.js (hunk 1 above)
Comment (shubhsherl) [reply app/lib/server/methods/deleteMessage.js:35]: So if you delete any offline message, if the app is connected (like attachment, which is still uploading) and you happen to delete it. then you get a toast showing the error, which is unnecessary.

File: app/lib/server/methods/sendMessage.js (hunk 2 above)
Comment (shubhsherl) [reply app/lib/server/methods/sendMessage.js:25]: For an offline message, we need to allow the message, even if the `ts` is greater than 60000.

File: app/message-attachments/client/messageAttachment.js (hunk 3 above)
Comment (shubhsherl) [reply app/message-attachments/client/messageAttachment.js:44]: Probably upstream has added a template between message and message attachment. To access message data in messageAttachment.js, like here https://github.com/WideChat/Rocket.Chat/blob/f92f50cedbe457dfb723cc04122d29e95362e671/app/message-attachments/client/messageAttachment.js#L84

File: app/lib/server/methods/deleteMessage.js (hunk 1 above)
Comment (Shailesh351) [reply app/lib/server/methods/deleteMessage.js:35]: Okay sounds good. Can you add same comments in the code? For future reference Thanks 😄

File: app/lib/server/methods/sendMessage.js (hunk 2 above)
Comment (Shailesh351) [reply app/lib/server/methods/sendMessage.js:25]: Same here. Add same comment in the code 😄

File: app/message-attachments/client/messageAttachment.js (hunk 3 above)
Comment (Shailesh351) [reply app/message-attachments/client/messageAttachment.js:44]: Cool. Looks good. May be in the upstream catch up it will be updated

**Comment by shubhsherl** [comment]
@Shailesh351 @ear-dev updated the required changes in #210
DO NOT MERGE THIS PR.
Merge PR: 210 as it includes the changes from this PR too(required for cypress test).

**Comment by ear-dev** [comment]
@shubhsherl I tend to think it would be better to keep the PRs separate and distinct.  What do you think about removing the double attachment fix from PR 210, we finish this review, and get it merged.  Then you can catch up to develop_pwa in PR 210 and it should be good to go.  @Shailesh351 do you agree?

In general smaller PRs are better, and each PR should address a single issue / feature or change if possible.

**Comment by ear-dev** [comment]
@Shailesh351 when you think this PR is ready please hit the approve button so that I know there is no more discussion going on and that we can merge it. Do you have permissions for that?  Thanks. 

image

**Comment by Shailesh351** [comment]
@ear-dev Actually as @shubhsherl mentioned this changes are already in https://github.com/WideChat/Rocket.Chat/pull/210. So I have reviewed that one.

**Comment by ear-dev** [comment]
ok, can we close this one then?

**Comment by shubhsherl** [comment]
Merged in #210
